extensions/binding/org.eclipse.smarthome.binding.homematic.test/src/test/java/org/eclipse/smarthome/binding/homematic/internal/communicator/virtual/ButtonDatapointTest.java

```diff
@@ -0,0 +1,159 @@
+
+        mockEventReceiver.eventReceived(longPressDp);
+
+        assertThat(buttonVirtualDatapoint.getValue(), is(CommonTriggerEvents.LONG_PRESSED));
+    }
+
+    @Test
+    public void testUnsupportedEvents() throws IOException, HomematicClientException {
+        HmDatapoint contPressDp = createPressDatapoint("PRESS_CONT", Boolean.TRUE);
+        HmDatapoint contButtonVirtualDatapoint = contPressDp.getChannel().getDatapoints().get(0);
+
+        mockEventReceiver.eventReceived(contPressDp);
+
+        HmDatapoint releaseDp = createPressDatapoint("PRESS_LONG_RELEASE", Boolean.TRUE);
+        HmDatapoint releaseButtonVirtualDatapoint = releaseDp.getChannel().getDatapoints().get(0);
+
+        mockEventReceiver.eventReceived(releaseDp);
+
+        HmDatapoint crapDp = createPressDatapoint("CRAP", Boolean.TRUE);
+        HmDatapoint crapButtonVirtualDatapoint = releaseDp.getChannel().getDatapoints().get(0);
+
+        mockEventReceiver.eventReceived(crapDp);
+
+        assertThat(contButtonVirtualDatapoint.getValue(), nullValue());
+        assertThat(releaseButtonVirtualDatapoint.getValue(), nullValue());
+        assertThat(crapButtonVirtualDatapoint.getValue(), nullValue());
+    }
+
+    @Test
+    public void testDoublePress() throws IOException, HomematicClientException, InterruptedException {
+        HmDatapoint shortPressDp = createPressDatapoint("PRESS_SHORT", Boolean.TRUE);
+        HmDatapoint buttonVirtualDatapoint = shortPressDp.getChannel().getDatapoints().get(0);
+
+        mockEventReceiver.eventReceived(shortPressDp);
+        assertThat(buttonVirtualDatapoint.getValue(), is(CommonTriggerEvents.SHORT_PRESSED));
+
+        Thread.sleep(DISABLE_DATAPOINT_DELAY / 2);
+
+        shortPressDp.setValue(Boolean.TRUE);
+        mockEventReceiver.eventReceived(shortPressDp);
+        assertThat(buttonVirtualDatapoint.getValue(), is(CommonTriggerEvents.DOUBLE_PRESSED));
+
+        Thread.sleep(DISABLE_DATAPOINT_DELAY * 2);
+
+        shortPressDp.setValue(Boolean.TRUE);
+        mockEventReceiver.eventReceived(shortPressDp);
+        assertThat(buttonVirtualDatapoint.getValue(), is(CommonTriggerEvents.SHORT_PRESSED));
+    }
+
+    private HmDatapoint createPressDatapoint(String channelName, Object value) {
+        HmDatapoint pressDp = new HmDatapoint(channelName, "", HmValueType.ACTION, value, true, HmParamsetType.VALUES);
+        HmChannel hmChannel = new HmChannel(channelName, 1);
+        // HmDatapoint buttonVirtualDatapoint = new HmDatapoint(HomematicConstants.VIRTUAL_DATAPOINT_NAME_BUTTON, "",
+        // HmValueType.ACTION, Boolean.FALSE, true, HmParamsetType.VALUES);
+        HmDevice device = new HmDevice("ABC12345", HmInterface.RF, "HM-MOCK", "mockid", "mockid", "mockfw");
+        hmChannel.setDevice(device);
+        device.addChannel(hmChannel);
+        bvdpHandler.initialize(device);
+        // shortPressChannel.addDatapoint(buttonVirtualDatapoint);
```

Comment:
Here are three lines commented out, should they have been removed?

Reply:
That is correct. I decided to use `bvdpHandler.initialize(device);` instead of initializing it manually. But I forgot to remove it. Thx